plugins/suggested-blocks/src/index.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * @license
+ * Copyright 2012 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Utility functions for handling suggestions.
+ */
+'use strict';
+
+/**
+ * Utility functions for handling block suggestions.
+ * @namespace Blockly.SuggestedBlocks
+ */
+import * as Blockly from 'blockly';
+
+const NUM_BLOCKS_PER_CATEGORY = 10;
+
+// Map from workspace ID to BlockSuggestor objects
+const suggestorLookup = new WeakMap();
+
+/**
+ *
+ */
+export class BlockSuggestor {
+  /**
+   * Constructs a BlockSuggestor object.
+   */
+  constructor() {
+    this.defaultJsonForBlockLookup = {};
+    this.recentlyUsedBlocks = [];
+    this.workspaceHasFinishedLoading = false;
+
+    this.eventListener = this.eventListener.bind(this);
+    this.getMostUsed = this.getMostUsed.bind(this);
+    this.getRecentlyUsed = this.getRecentlyUsed.bind(this);
+  }
+
+  /**
+   * Generates a list of the 10 most frequently used blocks, in order.
+   * Includes a secondary sort by most recent blocks.
+   * @return {Array <object>}A list of block JSON objects
```

Comment:
I think you might be able to do `!Array<!Blockly.utils.toolbox.BlockInfo>` ([link](https://github.com/google/blockly/blob/61322294da70c770b32abb28b9d695af75f61a16/core/utils/toolbox.js#L52), check that it's defined before commiting) which is a more specific type. Here and below.